matlab_simulation/05-estimation/kalman_filter/kf_example1.m

```diff
@@ -0,0 +1,133 @@
+% Kalman Filter Example 1: Temperature Control
+% 'Estimation 1' topic slide 48
+% State : Current temperature difference with outside
+% Motion : Decaying temperature + furnace input + disturbances
+% Measurement : Current temperature difference
+% Controller : Bang bang controller
+
+clear all
+clc
+close all
+
+% Example number
+example = 1;
+
+% Discrete time step
+dt = 0.1;
+
+% Prior
+mu = 10; % mean (mu)
+S = 1;   % covariance (Sigma)
+
+% Get matrices for motion model and measurement model
+[A,B,R,n] = motion_model(example,dt);
+[C,D,Q,m] = measurement_model(example);
+
+% Store in a structure
+ssm.A = A;
+ssm.B = B;
+ssm.C = C;
+ssm.D = D;
+ssm.R = R;
+ssm.Q = Q;
+ssm.n = n;
+ssm.m = m;
+
+% Simulation Initializations
+Tf = 3;
+T = 0:dt:Tf;
+x = zeros(1,length(T)+1);
+x(1) = mu+sqrt(S)*randn(1);
+y = zeros(m,length(T));
+u = y;
+
+%% Main loop
+for t=1:length(T)
+    % Simulation ----------------------------------------------------
+    % Select control action
+    if (t>1)
+        u(t)=u(t-1);
+    end
+    if (mu > 10)
+        u(t) = 0;
+    elseif (mu < 2);
+        u(t) = 1;
+    end
+    
+    % Select a motion disturbance
+    e = sqrt(R)*randn(1);
+    
+    % Update state
+    x(t+1) = A*x(t)+ B*u(t) + e;
+
+    % Take measurement
+    % Select a measurement disturbance
+    d = sqrt(Q.Q)*randn(1);
+    
+    % Determine measurement
+    y(t) = C*x(t+1) + d;
+    
+    % Store prior --------------------------------------------------
+    mu_old = mu;
+    S_old = S;
+    
+    % Kalman Filter Estimation
+    [mu,S,mup,Sp,K] = kalman_filter(ssm,mu,S,u(:,t),y(:,t),example,t);
+    
+    % Store estimates
+    mup_S(t)= mup;
+    mu_S(t)= mu;
+    
+    % Plot first time step results
+    if (t == 1)
+        L = 5;
+        
+        % Prior belief
+        figure(1);clf; subplot(2,2,1); hold on;
+        z = [mu_old-L*sqrt(S_old):0.01:mu_old+L*sqrt(S_old)];
```

Comment:
I would use unique names for plotting and not reuse z. I recommend renaming subsequent uses of z into new names.